addons/account/models/account_move.py

```diff
@@ -4482,6 +4482,21 @@ def _generate_pdf_and_send_invoice(self, template, force_synchronous=True, allow
         composer = self.env['account.move.send'].create(composer_vals)
         return composer.action_send_and_print(force_synchronous=force_synchronous, allow_fallback_pdf=allow_fallback_pdf, bypass_download=bypass_download)
 
+    def _get_portal_download(self):
+        """ Return existing attachments or a temporary Pro Forma pdf. """
+        if self.invoice_pdf_report_id:
+            attachments = self.env['account.move.send']._get_invoice_extra_attachments(self)
+        else:
```

Comment:
```suggestion
        elif: self.message_main_attachment_id:
            return self.message_main_attachment_id
        else:
```
In most cases from the portal view, the Send & Print should already have run before. If it fails it generates and assigns the proforma to `message_main_attachment_id`, so let's take advantage of that and avoid to re-generate it if it exists :slightly_smiling_face:

Reply:
Makes sense, so the method below (`get_invoice_pdf_report_attachment`) is just wrong ? :grimacing:

Reply:
Yes... well it's outside of the scope of this fix but I'll have a closer look :sweat_smile: